Sources/Exporters/OpenTelemetryProtocol/trace/utils/HTTPClient.swift

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import Foundation
+
+/// Client for sending requests over HTTP.
+internal final class HTTPClient {
+    private let session: URLSession
+
+    convenience init() {
+        let configuration: URLSessionConfiguration = .ephemeral
+        // NOTE: RUMM-610 Default behaviour of `.ephemeral` session is to cache requests.
+        // To not leak requests memory (including their `.httpBody` which may be significant)
+        // we explicitly opt-out from using cache. This cannot be achieved using `.requestCachePolicy`.
+        configuration.urlCache = nil
+        // TODO: RUMM-123 Optimize `URLSessionConfiguration` for good traffic performance
+        // and move session configuration constants to `PerformancePreset`.
+        self.init(session: URLSession(configuration: configuration))
+    }
+
+    init(session: URLSession) {
+        self.session = session
+    }
+
+    func send(request: URLRequest, completion: @escaping (Result<HTTPURLResponse, Error>) -> Void) {
+        let task = session.dataTask(with: request) { data, response, error in
```

Comment:
It should not, but the traces will appear. There is also a callback function in the configuration that can be used to exclude the internal URL's for reporting: `shouldInstrument`, and return `false` for the URLs we want to exclude

Reply:
Do we need to update documentation at a minimum to reflect this?

Or a better solution may be to add a custom header into the HTTPClient, such as `X-OpenTelemetry-SDK: swift` (random example), check for this within the URLSession instrumentation and filter it out?

We've had problems before with instrumentation instrumenting itself, so know this is easy to have accidentally happen!

Reply:
> Do we need to update documentation at a minimum to reflect this?
Yes, documentation is currently very basic, what more things you think that should be clarified? I wrote the current one, and I thought it was short  but clear, thats why I'd like some feedback

> Or a better solution may be to add a custom header into the HTTPClient, such as `X-OpenTelemetry-SDK: swift` (random example), check for this within the URLSession instrumentation and filter it out?
I would prefer not to add extra headers just to identify these requests, since other internal requests might also be filtered by the user, I think clarifying how to filter is a better option